microsoft/testsuites/dpdk/common.py

```diff
@@ -17,6 +22,249 @@
+        else:
+            self.tar_filename = PurePath(self._tar_url).name
+            remote_path = work_path.joinpath(self.tar_filename)
+            node.shell.copy(
+                local_path=PurePath(self._tar_url),
+                node_path=remote_path,
+            )
+        # create tarfile dest dir
+        self._asset_path = work_path.joinpath(
+            self.tar_filename[: -(len(tarfile_suffix))]
+        )
+        # unpack into the dest dir
+        # force name as tarfile name
+        node.tools[Tar].extract(
+            file=str(remote_path),
+            dest_dir=str(work_path),
+            gzip=True,
+        )
+        return self._asset_path
+
+
+class Installer:
+    # Generic 'Installer' parent class for DpdkTestpmd/rdma-core
+    # NOTE: This should not be instantiated directly.
+    _err_msg = "not implemented for this installation type."
+
+    # setup the node before starting
+    # ex: updating the kernel, enabling features, checking drivers, etc.
+    def _setup_node(self) -> None:
+        raise NotImplementedError(f"_setup_node {self._err_msg}")
+
+    # check if the package is already installed:
+    # Is the package installed from source? Or from the package manager?
+    # Does the version match the one we want if we need a specific one?
+    def _check_if_installed(self) -> bool:
+        raise NotImplementedError(f"_check_if_installed {self._err_msg}")
+
+    # setup the installation (install Ninja, Meson, etc)
+    def _download_assets(self) -> None:
+        if self._downloader:
+            self._asset_path = self._downloader.download()
+        else:
+            self._node.log.debug("No downloader assigned to installer.")
+
+    # do the build and installation
+    def _run_build(self) -> None:
+        self._download_assets()
+
+    # remove an installation
+    def _uninstall(self) -> None:
+        raise NotImplementedError(f"_clean_previous_installation {self._err_msg}")
+
+    # install the dependencies
+    def _install_dependencies(self) -> None:
+        if self._os_dependencies is not None:
+            self._os_dependencies.install_required_packages(
+                self._node, extra_args=self._package_manager_extra_args
+            )
+
+    # define how to check the installed version
+    def get_installed_version(self) -> VersionInfo:
+        raise NotImplementedError(f"get_installed_version {self._err_msg}")
+
+    def _should_install(self, required_version: Optional[VersionInfo] = None) -> bool:
+        return (not self._check_if_installed()) or (
+            required_version is None and required_version > self.get_installed_version()
+        )
+
+    # run the defined setup and installation steps.
+    def do_installation(self, required_version: Optional[VersionInfo] = None) -> None:
+        self._setup_node()
+        if self._should_install():
+            self._uninstall()
+            self._install_dependencies()
+            self._run_build()
```

Comment:
How about call it `_install`? so it's consistent with `_uninstall`. The `do_installation` can be called like `install`. The public/private methods is a pattern in LISA, like `initialize`/`_initialize`. The public method is used to be called externally, and the private method is implemented by subclasses.

Reply:
Ok